ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/parts/PluginVersionPart.java

```diff
@@ -288,6 +288,38 @@ private void createSingleField(Composite parent, boolean createGroup, boolean ed
 		fMinVersionText = new Text(parent, SWT.SINGLE | SWT.BORDER);
 		fMinVersionText.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
 		fMinVersionText.setEnabled(editable);
+
+		if (!fRangeAllowed) {
+			Label incrLabel = new Label(parent, SWT.NONE);
+			incrLabel.setText(PDEUIMessages.PluginVersionPart_incrementTitle);
+			incrLabel.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, false, false));
+
+			Composite btnBar = new Composite(parent, SWT.NONE);
+			btnBar.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+
+			GridLayout btnBarLayout = new GridLayout(3, true);
+			btnBarLayout.marginWidth = 0;
+			btnBar.setLayout(btnBarLayout);
+
+			Button majorBtn = new Button(btnBar, SWT.PUSH);
+			majorBtn.setText(PDEUIMessages.PluginVersionPart_bumpMajor);
+			majorBtn.setEnabled(editable);
+			majorBtn.addListener(SWT.Selection, e -> bumpVersion(Type.MAJOR));
```

Comment:
Please use a typed-listener here, i.e. call `addSelectionListener()` instead and create a `SelectionListener` using `SelectionListener.widgetSelectedAdapter()`.
Together with the suggestion below to pass a lambda, this code would be
```suggestion
			majorBtn.addSelectionListener(SelectionListener.widgetSelectedAdapter(e -> bumpVersion(v -> {
				return new Version(v.getMajor() + 1, 0, 0, v.getQualifier());
			})));
```
The other button definitions would looks correspondingly.
I'm not sure if it's necessary to handle the qualifier here. In general versions of exported-packages don't have a qualifier anyway. But it could also be handled uniformly in the `bumpVersion()` (if really necessary).